bitcoin/exchange/bitcoinprices.go

```diff
@@ -17,14 +16,27 @@ const SatoshiPerBTC = 100000000
 
 var log = logging.MustGetLogger("exchangeRates")
 
-type ExchangeRateProvider interface {
-	fetch() error
+type ExchangeRateProvider struct {
+	fetchUrl string
+	cache map[string]float64
```

Comment:
no need for pointer as map is a reference type, so I learned.
(still learning golang)